File: 2024/go/main.go
package main

import (
	"aoc/puzzles"
	"fmt"
	"time"
)

func main() {
	currentDate := time.Now().Format("02-01-2006")
	fmt.Printf("AoC: %s\n", currentDate)
	fmt.Println("-----------------")

	// test boilerplate
	// Note - Go does not suport default values or optional parameters.
	// testDate := "01-12-2000"
	// params: date, useTestData
	// puzzles.Boilerplate.SolvePart1(testDate, true)
	// puzzles.Boilerplate.SolvePart2(testDate, true)

	// run puzzle solvers
	// puzzles.Dec01.SolvePart1(currentDate, false) // ans: 3508942
	// puzzles.Dec01.SolvePart2(currentDate, false) // ans: 26593248

	// puzzles.Dec02.SolvePart1(currentDate, true) // ans:
	// puzzles.Dec02.SolvePart2(currentDate, true) // ans:

	// puzzles.Dec03.SolvePart1(currentDate, true) // ans:
	// puzzles.Dec03.SolvePart2(currentDate, true) // ans:

	// puzzles.Dec04.SolvePart1("04-12-2024", false) // ans: 2464
	puzzles.Dec04.SolvePart2("04-12-2024", false) // ans: 1982
}

File: 2024/python/main.py
from datetime import datetime

from puzzles import (
	boilerplate,
	dec01,
	dec02,
	dec03,
	dec04,
	dec05
)


def main():
	current_date = datetime.now().strftime("%d-%m-%Y")
	
	print(f"AoC: {current_date}")
	print('-----------------\n')

	# test boilerplate
	# test_date = "01-12-2000"
	# boilerplate.solve_part1(test_date)
	# boilerplate.solve_part2(test_date, use_test_data=False)

	# Run puzzle solvers
	# dec01.solve_part1(current_date, use_test_data=False) # ans: 3508942
	# dec01.solve_part2(current_date, use_test_data=False) # ans: 26593248


	# dec02.solve_part1("02-12-2024", use_test_data=False) # ans: 282
	# dec02.solve_part2("02-12-2024", use_test_data=False) # ans: 349

	# dec03.solve_part1(current_date, use_test_data=True) # ans: 188192787
	# dec03.solve_part2(current_date, use_test_data=True) # ans: 113965544

	# dec04.solve_part1("04-12-2024", use_test_data=False) # ans: 2464
	# dec04.solve_part2("04-12-2024", use_test_data=False) # ans: 1982

	# dec05.solve_part1("04-12-2024", use_test_data=False) # ans: 6384
	# dec05.solve_part2("04-12-2024", use_test_data=False) # ans: 5353

# Run main function
if __name__ == "__main__":
	main()
